### PR TITLE
add documentation on EESSI test suite

### PR DESCRIPTION
**Pull request by boegel**
Work-in-progress, this page is still being actively developed via https://hackmd.io/TUjhmb32SnertH9nD2Flaw

It is meant to eventually replace http://www.eessi.io/docs/software_testing, which is very outdated right now

### CODE REVIEW

File: docs/test-suite/usage.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+```
+$ reframe --list
+...
+[List of matched checks]
+- ...
+Found 123 check(s)
+```
+
+!!! note
+    When using `--list`, checks are only generated based on available modules.
```
Comment (smoors) [inline docs/test-suite/usage.md:25]:
```suggestion
    When using `--list`, checks are only generated based on modules that are available in the system where the `reframe` command is invoked.
```

File: docs/test-suite/usage.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+reframe --run --system example:cpu --name GROMACS --tag 4_cores --performance-report
+```
+
+### List all checks for TensorFlow 2.11 using a single node
+
+```
+reframe --list --name %module_name=TensorFlow/2.11 --tag 1_node
+```
+
+### Dry run of TensorFlow CI checks on a quarter node (on all system partitions)
```
Comment (smoors) [inline docs/test-suite/usage.md:244]:
```suggestion
### Dry run of TensorFlow CI checks on a quarter (1/4) of a node (on all system partitions)
```

**Comment by smoors** [comment]
should we also say something about multi-threading?

File: docs/test-suite/usage.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+#### `scale` tags
+
+The EESSI test suite defines a set of custom tags that control the *scale* of checks,
+which specify many cores/nodes should be used for running a check.
+
+| tag name | description |
+|:--------:|-------------|
+| `1_core` | using a single CPU core, or single GPU |
+| `2_cores` | using 2 CPU cores, or 2 GPUs |
+| `4_cores` | using 4 CPU cores, or 4 GPUs |
```
Comment (smoors) [inline docs/test-suite/usage.md:193]: this is not entirely correct
```suggestion
| `1_core` | using 1 CPU core and 1 GPU (if running a GPU test) |
| `2_cores` | using 2 CPU cores and 1 GPU (if running a GPU test) |
| `4_cores` | using 4 CPU cores, or 4 GPUs and 1 GPU (if running a GPU test) |
```

**Comment by boegel** [comment]
@smoors All suggestions applied, good to go now?

File: docs/test-suite/usage.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+#### `scale` tags
+
+The EESSI test suite defines a set of custom tags that control the *scale* of checks,
+which specify many cores/nodes should be used for running a check.
+
+| tag name | description |
+|:--------:|-------------|
+| `1_core` | using 1 CPU core and 1 GPU (if running a GPU test) |
+| `2_cores` | using 2 CPU cores and 1 GPU (if running a GPU test) |
+| `4_cores` | using 4 CPU cores, or 4 GPUs and 1 GPU (if running a GPU test) |
```
Comment (smoors) [inline docs/test-suite/usage.md:193]:
```suggestion
| `4_cores` | using 4 CPU cores and 1 GPU (if running a GPU test) |
```

**Comment by smoors** [comment]
> @smoors All suggestions applied, good to go now?

almost, just one fix (see above)

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
all good for me, but i don't have merge rights

**Review by casparvl — CHANGES_REQUESTED** [review]

**Review by casparvl — APPROVED** [review]
Lgtm!

**Pull request merged**